src/shared/generator.c

```diff
@@ -82,7 +82,7 @@ static int write_fsck_sysroot_service(
                 const char *what,
                 const char *extra_after) {
 
-        _cleanup_free_ char *device = NULL, *escaped = NULL, *escaped2 = NULL;
+        _cleanup_free_ char *device = NULL, *escaped = NULL, *escaped2 = NULL, *makefs_unit;
```

Comment:
needs NULL assignment, otherwise might be destroyed in an early error path without ever being initialized → memory corruption

Reply:
Ah, whoops. Fixed.